src/controllers/hygrometry_regulation_controller.py

```diff
@@ -0,0 +1,170 @@
+        self._cummulative = [0.0]*16
+        self._last_read = [0.0]*16
+        self._average = [0.0]*16
+        self._nb_sample = [0]*16
+        self._interval_update = hygro_config[_INTERVAL_UPDATE]
+        self._delta_detection = hygro_config[_DELTA_DETECTION]
+        self._max_sample_regulation = hygro_config[_MAX_SAMPLE_REGULATION]
+        self._previous_read_time = 0
+
+        # water Shot
+        self._shot_duration = hygro_config[_SHOT_DURATION]
+        self._pump_speed = hygro_config[_PUMP_SPEED]
+        self._shot_query_queue = []  # Contains plant position line.
+        self._query_status = False # False:wating, True:In progress
+        self._previous_shot_time = 0
+
+        # Initialisation
+        for i in range(config[_PLANT_COUNT]):
+            hygro_value = ADCService.instance().get_plant_hygrometry_value(i)
+            self._cummulative[i] = hygro_value
+            self._last_read[i] = hygro_value 
+            self._average[i] = hygro_value
+            self._nb_sample[i] = 1
+
+    def update(self, plants):
+        """
+        Execute the following updates:
+        - Check for add or removed plants
+        - Check for Hygrometry
+        - Give some water shots for hygrometry regulation
+        :param plants: plant list - 16 elements by ASC position
+        :rtype plants: list
+        """
+        self._shot_update()
+
+        if time_in_millisecond() - self._previous_read_time < self._interval_update:
+            self._hygrometric_update(plants)
+            self._previous_read_time = time_in_millisecond()
+
+    def _hygrometric_update(self, plants):
+        """
+        Check for the added or removed plants and ask for hygrometric regulation
+        :param plants: plant list - 16 elements by ASC position but it doesn't matter
+        :rtype plants: list
+        """
+        for plant in plants: 
+            i = plant.position()
+            hygro_val = ADCService.instance().get_plant_hygrometry_value(i)
+
+            # Si on a détecté une diférence d'hygrométrie spontannée on conjuge avec la dernière mesure 
+            # pour confirmer ce changement et détecter l'ajout ou le retrait d'un pot.
+            # if hygro Val != avergae +/- delta AND hygro_val == last_read +/- delta
+            if ((hygro_val <= (self._average(i) - self._delta_detection)) || (hygro_val >= (self._average(i) + self._delta_detection))) && 
+            ((hygro_val >= (self._last_read(i) - self._delta_detection)) && (hygro_val <= (self._last_read(i) + self._delta_detection))):
+
+                # If it was an adding
+                if hygro_val >= (self._average(i) + self._delta_detection)
+                    # Ajouter dans la dB
+                    DatabaseService.instance().execute(INSERT_NEW_PLANT, i)
+                else
+                    # Retirer de la DB
+                    DatabaseService.instance().execute(REMOVE_PLANT, plants(i).uuid(), i)

```

Comment:
```suggestion
                else:
                    # Retirer de la DB
                    DatabaseService.instance().execute(REMOVE_PLANT, plants[i].uuid, i)

```

uuid est une propriete de `Plant` pas besoin de mettre de parentheses